### ISSUE TITLE
google_compute_firewall target_tags and source_tags (may be more widespread) cannot be interpolated

### ISSUE THREAD

**Issue by wendorf** [hashicorp/terraform#11063]
### Terraform Version
Terraform v0.8.2

This also is happening as of Terraform v0.8.3-dev (f4c5b9fadae241533594f5d8bcb86339402c7a30)

### Affected Resource(s)
- google_compute_firewall

Could very well be others, though, since the problem commit is not in the Google provider

### Terraform Configuration Files
```hcl
variable "project" {
  type = "string"
}

variable "region" {
  type = "string"
}

variable "service_account_key" {
  type = "string"
}
variable "env_name" {
  type = "string"
}

provider "google" {
  project     = "${var.project}"
  region      = "${var.region}"
  credentials = "${var.service_account_key}"
}

resource "google_compute_network" "network" {
  name = "${var.env_name}-network"
}

// Allow open access between internal VMs for a PCF deployment
resource "google_compute_firewall" "firewall" {
  name       = "${var.env_name}-firewall"
  network    = "${google_compute_network.network.name}"

  allow {
    protocol = "icmp"
  }

  target_tags = ["my-target-tag"]

  source_tags = ["my-source-tag"]
}

output "my-source-tags" {
  value = "${google_compute_firewall.firewall.source_tags}"
}

output "my-target-tags" {
  value = "${google_compute_firewall.firewall.target_tags}"
}
```

### Debug Output
https://gist.github.com/wendorf/3deaa230d6b9c48c44639a16d135d67b

### Expected Behavior
The outputs for `my-source-tags` and `my-target-tags` should contain the `my-target-tag` and `my-source-tag` values we specified in google_compute_network.network.

```
Outputs:

my-source-tags = [
    my-source-tag
]
my-target-tags = [
    my-target-tag
]
```

### Actual Behavior
The outputs have the correct number of tags, but the tags themselves are empty strings.

```
Outputs:

my-source-tags = [

]
my-target-tags = [

]
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
We did a `git bisect` to determine that https://github.com/hashicorp/terraform/commit/084670c2417196e76d2ed73ba1d93ea4e0788d24 is the commit that broke the behavior.

**Comment by mitchellh** [comment]
Fix in #11042, will be in the next release.

**Comment by lobeck** [comment]
Still got issues with 0.8.6 and a case like:

    resource "google_compute_instance" "vpn" {
        tags = "${google_compute_firewall.vpn.target_tags}"
    }

results in:

    google_compute_instance.vpn #0: tags: should be a list

also casts to `list()` don't change anything.

Is this related?
